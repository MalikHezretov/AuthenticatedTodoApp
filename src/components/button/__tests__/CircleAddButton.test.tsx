import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react-native';
import CircleAddButton from '..';

const mockFn = jest.fn();
test('should render without error', () => {
  const tree = render(<CircleAddButton onPressAddButton={mockFn} />);

  expect(tree).toMatchSnapshot();
});

test('Press add card button', () => {
  render(<CircleAddButton onPressAddButton={mockFn} />);

  fireEvent.press(screen.getByText('Add'));

  expect(mockFn).toBeCalled();
});
