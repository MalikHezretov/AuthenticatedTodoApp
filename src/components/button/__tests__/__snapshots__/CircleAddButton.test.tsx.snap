// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without error 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#0096FF",
      "borderRadius": 40,
      "height": 80,
      "justifyContent": "center",
      "opacity": 1,
      "width": 80,
    }
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontSize": 24,
      }
    }
  >
    Add
  </Text>
</View>
`;
